AristaLibrary/Expect.py

```diff
@@ -392,6 +392,48 @@ def refresh_command_output_on_devices(self, cmd=None):
         """
         return self.get_command_output(cmd=cmd)
 
+    def record_output(self, switch_id=None, cmd=None, encoding='text'):
+        """Log the provided command. If no command is provided then log the
+        contents of the currently stored command result. Default is to log
+        text format of a command output.
+
+        Args:
+            switch_id (int, optional): The index id for a specific switch
+                connection defined in the AristaLibrary. If not specified,
+                all available switch connections will be used in sequence.
+            cmd (string, optional): The command string that will be exectuted
+                on the switch or switches determined by switch_id. If not
+                specified, the previous command sent to each switch will
+                be reused, or the command used in the library import if
+                no previous command has been sent. Default is None.
+            encoding (string, default=text): The format in which the command
+                output should be logged. Options are text or json. Not all
+                commands are able to return json formatted output and in this
+                case the output will automatically be returned in text format.
+        """
+        # Convert the passed in switch_id to the actual switch info dict
+        # Use current switch if switch_id is not specified.
+        if switch_id:
+            switch = self.arista_lib.get_switch(switch_id)
+        else:
+            switch = self.arista_lib.get_switch()
+
+        result = None
+        if cmd:
+            if encoding not in ['text', 'json']:
+                encoding = 'text'
```

Comment:
I like where you went with a sane default but should we actually return an error if an invalid value was passed?

Reply:
My thought is that a user will most of the time want text even if they typo it and won't necessarily think about what the valid options are for encoding. I also prefer to avoid failing a test case because of a logging error.

Reply:
Thanks.   Ok.